lib/node_modules/@stdlib/string/len/lib/should-break.js

```diff
@@ -0,0 +1,165 @@
+var BreakLastRegional = require( './constants.js' ).BreakLastRegional;
+var BreakPenultimateRegional = require( './constants.js' ).BreakPenultimateRegional;
+var ExtendedPictographic = require( './constants.js' ).Extended_Pictographic;
+
+
+// MAIN //
+
+/**
+* Returns whether a break is allowed between the two given grapheme breaking classes according to UAX #29 3.1.1 Grapheme Cluster Boundary Rules on extended grapheme clusters.
+*
+* @param {number} start - start
+* @param {Array<number>} mid - mid
+* @param {number} end - end
+* @param {number} startEmoji - startEmoji
+* @param {Array<number>} midEmoji - midEmoji
+* @param {number} endEmoji - endEmoji
+* @returns {number} break type
+*/
+function shouldBreak( start, mid, end, startEmoji, midEmoji, endEmoji ) {
+	var prevNonExtendIdx;
+	var regIndRange;
+	var nextEmoji;
+	var allEmoji;
+	var previous;
+	var next;
+	var all;
+	/**
+	* Function expression.
+	* @private
+	* @param {number} c - temp
+	* @returns {boolean} - check if extend
+	*/
+	function checker( c ) {
+		return c === RegionalIndicator;
+	}
+	/**
+	* Function expression.
+	* @private
+	* @param {number} c - temp
+	* @returns {boolean} - check if Extend
+	*/
+	function checkerE( c ) {
+		return c === Extend;
+	}
+	all = [ start ].concat( mid ).concat( [ end ] );
+	allEmoji = [ startEmoji ].concat( midEmoji ).concat( [ endEmoji ] );
+	previous = all[ all.length - 2 ];
+	next = end;
+	nextEmoji = endEmoji;
+
+	regIndRange = all.lastIndexOf( RegionalIndicator );
+	if ( regIndRange > 0 &&
+		all.slice( 1, regIndRange ).every( checker ) &&
+		[ Prepend, RegionalIndicator ].indexOf( previous ) === -1 ) {
+		if ( all.filter( checker ).length % 2 === 1 ) {
+			return BreakLastRegional;
+		}
+		return BreakPenultimateRegional;
+	}
+
+	// GB3: CR × LF
+	if ( previous === CR && next === LF ) {
+		return NotBreak;
+	}
+	// GB4: (Control|CR|LF) ÷
+	else if ( previous === Control || previous === CR || previous === LF ) { // eslint-disable-line no-else-return
+		return BreakStart;
+	}
+	// GB5: ÷ (Control|CR|LF)
+	else if ( next === Control || next === CR || next === LF ) {
+		return BreakStart;
+	}
+	// GB6: L × (L|V|LV|LVT)
+	else if ( previous === L &&
+		( next === L || next === V || next === LV || next === LVT ) ) {
+		return NotBreak;
+	}
+	// GB7: (LV|V) × (V|T)
+	else if ( (previous === LV || previous === V ) &&
+		( next === V || next === T ) ) {
+		return NotBreak;
+	}
+	// GB8: (LVT|T) × (T)
+	else if ( ( previous === LVT || previous === T ) &&
+		next === T ) {
+		return NotBreak;
+	}
+	// GB9: × (Extend|ZWJ)
+	else if ( next === Extend || next === ZWJ ) {
+		return NotBreak;
+	}
+	// GB9a: × SpacingMark
+	else if ( next === SpacingMark ) {
+		return NotBreak;
+	}
+	// GB9b: Prepend ×
+	else if ( previous === Prepend ) {
+		return NotBreak;
+	}
+
+	// GB11: \p{Extended_Pictographic} Extend* ZWJ × \p{Extended_Pictographic}
+	prevNonExtendIdx = allEmoji.slice( 0, -1 ).lastIndexOf( ExtendedPictographic ); // eslint-disable-line max-len
```

Comment:
Rather than perform a slice here, would be good to have a small utility function which performs the same operation as `lastIndexOf` but accepts an `N` parameter specifying the number of values over which to iterate. In which case, here, instead of `N`, we'd provide `N-1`, thus excluding the last array element from consideration. For examples, see the strided array functionality in `@stdlib/stats/base/*`.

Reply:
Sorry for missing it earlier, but [lastIndexOf](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/lastIndexOf) itself supports a `fromIndex` argument. I will just pass `N - 2` as the `fromIndex` argument. I hope that is fine.